libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/NumberInput.cs

```diff
@@ -109,17 +108,8 @@ protected override Task<InputState> OnRecognizeInputAsync(DialogContext dc, Canc
 
         private string GetCulture(DialogContext dc)
         {
-            if (!string.IsNullOrEmpty(dc.Context.Activity.Locale))
-            {
-                return dc.Context.Activity.Locale;
-            }
-
-            if (this.DefaultLocale != null)
-            {
-                return this.DefaultLocale.GetValue(dc.State);
-            }
-
-            return English;
+            // Note: Default locale will be considered for deprecation as part of 4.13.
+            return dc.GetLocale() ?? DefaultLocale?.GetValue(dc.State) ?? string.Empty;
```

Comment:
This annoying line of code everywhere! Perhaps this DialogContext could know about some properties - its knows the active dialog right? and we could standardize on the DefaultLocale (discoverable via an interface on the dialog) and then hide this whole defaulting behavior behind the GetLocale function.

More abstractly, I wonder if there are any ideas  we can take from the C++ std library implementation, facets for example.

Reply:
We discussed some ideas on deprecating inputs specific default locale in r13 .